LibOS/shim/src/ipc/shim_ipc_nsimpl.h

```diff
@@ -527,8 +541,16 @@ int CONCAT3(del, NS, range) (IDTYPE idx)
 
     if (r->used)
         free(r->used);
-    hlist_del(&r->hlist);
-    list_del(&r->list);
+    // Re-acquire the head; kind of ugly
+    LISTP_TYPE(range) * head = range_table + RANGE_HASH(off);
+    listp_del(r, head, hlist);
+    /* Chia-Che Tsai 10/17/17: only when r->owner is non-NULL,
+     * and r->owner->vmid == cur_process.vmid, r is on the
+     * owned list, otherwise it is an offered. */
+    if (r->owner && r->owner->vmid == cur_process.vmid)
```

Comment:
@donporter Please check this change.

I believe the right logic is that only when owner is non-NULL and owner->vmid is not the current process, the range belongs to the owned list (owned by the current process). The following code should point out pretty clear.